sdk/go/common/apitype/core.go

```diff
@@ -326,6 +326,8 @@ type ResourceV3 struct {
 	ImportID resource.ID `json:"importID,omitempty" yaml:"importID,omitempty"`
 	// If set to True, the providers Delete method will not be called for this resource. Pulumi simply stops tracking the deleted resource.
 	RetainOnDelete bool `json:"retainOnDelete,omitempty" yaml:"retainOnDelete,omitempty"`
+	// An extra value to track, that if changed should trigger a replace.
+	Trigger interface{} `json:"trigger,omitempty" yaml:"trigger,omitempty"`
```

Comment:
One major question in adding this features is if this should be a new V4 state? If we don't then you could possibly lose the trigger state, although any SDK sending Trigger would of error'd before sending the RegisterResource message for it because "supports_features(trigger)" would of returned false so a normal update wouldn't cause this, but someone could run refresh or import on an old CLI and clobber the state.